src/controllers/v1/attachment/index.ts

```diff
@@ -71,6 +41,36 @@ export class attachment extends Controller {
     this.log = logger.child({ controller: '/attachment' })
   }
 
+  parseAccept = (acceptHeader: string) =>
+    acceptHeader
+      .split(',')
+      .map((acceptElement) => {
+        const trimmed = acceptElement.trim()
+        const [mimeType, quality = '1'] = trimmed.split(';q=')
+        return { mimeType, quality: parseFloat(quality) }
+      })
+      .sort((a, b) => {
+        if (a.quality !== b.quality) {
+          return b.quality - a.quality
+        }
+        const [aType, aSubtype] = a.mimeType.split('/')
+        const [bType, bSubtype] = b.mimeType.split('/')
+        if (aType === '*' && bType !== '*') {
+          return 1
+        }
+        if (aType !== '*' && bType === '*') {
+          return -1
+        }
+        if (aSubtype === '*' && bSubtype !== '*') {
+          return 1
+        }
+        if (aSubtype !== '*' && bSubtype === '*') {
+          return -1
+        }
+        return 0
+      })
+      .map(({ mimeType }) => mimeType)
+
   octetResponse(buffer: Buffer, name: string): Readable {
     // default to octet-stream or allow error middleware to handle
```

Comment:
I don't understand this comment 

`// default to octet-stream or allow error middleware to handle` ..

allow the middleware to handle errors .. where ? here ?

Reply:
```suggestion
    // default to octet-stream or allow error middleware to handle
```

```suggestion
    // this function defaults to octet-stream or it throws an error
```

Reply:
andy your suggestion remobved some code.